streamz/sources.py

```diff
@@ -35,15 +36,28 @@ def write(text):
 class Source(Stream):
     _graphviz_shape = 'doubleoctagon'
 
-    def __init__(self, **kwargs):
+    def __init__(self, start=False, **kwargs):
         self.stopped = True
-        super(Source, self).__init__(**kwargs)
+        super().__init__(ensure_io_loop=True, **kwargs)
+        self.started = False
+        if start:
+            self.start()
 
-    def stop(self):  # pragma: no cover
-        # fallback stop method - for poll functions with while not self.stopped
+    def stop(self):
+        """set self.stopped, which will cause polling to stop after next run"""
         if not self.stopped:
             self.stopped = True
 
+    def start(self):
+        """start polling"""
+        self.stopped = False
+        self.started = True
+        self.loop.add_callback(self.run)
+
+    async def run(self):
+        while not self.stopped:
+            await self._run()
```

Comment:
Should there be a `_run` method that raises `NotImplemented`?

Reply:
That would make sense